schema/table.go

```diff
@@ -372,7 +373,7 @@ func (t *Table) ValidateName() error {
 	if !ok {
 		return fmt.Errorf("table name %q is not valid: table names must contain only lower-case letters, numbers and underscores, and must start with a lower-case letter or underscore", t.Name)
 	}
-	return nil
+	return ValidateTable(t)
```

Comment:
There's some refactoring to be done around validation code as we have a separate `schema/validators.go` which also has some code to validate table names. It currently only validates the length, so I added the call here